### PR TITLE
Clean up Osteppy server

### PR DESCRIPTION
**Pull request by scrust**
Cleaned up a bunch of stuff in Osteppy:

- resolved lint issues per existing ESLint config
- allowed `console.log` statements as a NodeJS application
- cleaned up imports
- encapsulated EOD reporting as a class that manages in-memory data and persisting to files
  - did not reimplement `printRAs()` since it was a debugging function

### CODE REVIEW

File: Osteppy/src/server.js [hunk 1]
```diff
@@ -1,32 +1,97 @@
...
 import cors from 'cors';
 import morgan from 'morgan';
 import bodyParser from 'body-parser';
-import config from './config.json';
 import axios from 'axios';
-import * as fs from 'fs';
+import fs from 'fs';
+import config from './config.json';
+
+class EODList {
```
Comment (belavina) [inline Osteppy/src/server.js]: Would it make sense to split up the server & EODlist into 2 modules?

File: Osteppy/src/server.js [hunk 2]
```diff
@@ -1,32 +1,97 @@
...
+    return this.save();
+  }
+
+  missing() {
+    return [...this.unsubmittedNames];
+  }
+
+  report() {
+    return { ...this.reportData };
+  }
```
Comment (belavina) [inline Osteppy/src/server.js]: Just curious, why are you spreading in these 2 cases?

File: Osteppy/src/server.js [hunk 3]
```diff
@@ -1,32 +1,97 @@
...
-import config from './config.json';
 import axios from 'axios';
-import * as fs from 'fs';
+import fs from 'fs';
+import config from './config.json';
+
+class EODList {
+  constructor({
+    dataFile = path.join(__dirname, 'eods.json'),
+    unsubmittedNamesFile = path.join(__dirname, 'sleepyRAs.txt'),
```
Comment (belavina) [inline Osteppy/src/server.js]: Not really sure why we are using different file formats (maybe @naiuhz knows), I think it would make sense to use `.json` only (parsing `.txt` is more error prone)

File: Osteppy/src/server.js [hunk 4]
```diff
@@ -35,182 +100,124 @@ app.use(morgan('dev'));
...
+      channel: slackRequest.channel_name,
+    })).catch((error) => {
+      console.log(`error: ${error}`);
+    });
+
+  res.status(200).send();
 });
 
 // Slash command for checking who have yet to submit EOD's
 app.post('/eod_left', (req, res) => {
```
Comment (belavina) [inline Osteppy/src/server.js]: We could potentially group the requests e.g. `/eod`, `/eod/left`, `/eod/reminder/script_status`, `/eod/reminder/time` instead of  `/eod/eod_left`, `check_js_script` etc. (will need to update slack app configs then)

File: Osteppy/src/server.js (hunk 2 above)
Comment (scrust) [reply Osteppy/src/server.js]: My initial thinking was I wanted callers to avoid accidental data mutations, but since this is javascript and the data is accessible via `this`, it's probably unnecessary.

File: Osteppy/src/server.js (hunk 3 above)
Comment (scrust) [reply Osteppy/src/server.js]: `sleepyRA.txt` seems to just be a newline-delimited file so it's not hard to parse, but we could definitely transition everything to JSON.

File: Osteppy/src/server.js (hunk 1 above)
Comment (scrust) [reply Osteppy/src/server.js]: Sure, can definitely do this 👍

**Comment by scrust** [comment]
@belavina in light of [this comment](https://github.com/Seneca-CDOT/ostep-dashboard/pull/55#discussion_r288787320), do you know the purpose of `checkEODClock()`? It "Reads clock.txt to return the time", but why isn't e.g. `new Date()` or the system time sufficient?

**Comment by belavina** [comment]
@scrust I guess it serves some debugging purpose for the EODreminder bot?  @naiuhz probably knows. 
Looks like `server.js` forks a child process (`EODreminder.js`) & eodReminder writes its uptime to the txt file (so `server.js` knows its up & running or something (?)).

**Comment by scrust** [comment]
@poftadeh instead of refactoring `checkEODClock()` and `checkJSScript()` can we remove those endpoints? They seem to be Slackbot debugging commands, I'm not clear on their utility.

If we want to keep them, I think we should defer their refactor until someone looks at `EODreminder.js` since they seem to be pretty tied up in its implementation.

File: Osteppy/src/server.js (hunk 3 above)
Comment (scrust) [reply Osteppy/src/server.js]: Changing the filetype is also going to require modifying `EODreminder.js`, so it might be better to hold off and change the filetype everywhere as a separate PR.

Comment (belavina) [reply Osteppy/src/server.js]: okc :+1:

**Comment by poftadeh** [comment]
@scrust Let's defer refactoring it in that case, and address this in a future PR.
